fbw-a32nx/src/wasm/systems/a320_systems/src/hydraulic/sfcc/mod.rs

```diff
@@ -41,59 +49,44 @@ impl From<u8> for FlapsConf {
-        let tolerance = Angle::new::<degree>(Self::TARGET_THRESHOLD_DEGREE);
-        position > (target_position - tolerance) && position < (target_position + tolerance)
-    }
-
-    pub fn below_enlarged_target_range(position: Angle, target_position: Angle) -> bool {
-        let tolerance = Angle::new::<degree>(Self::ENLARGED_TARGET_THRESHOLD_DEGREE);
-        position < target_position - tolerance
-    }
-
-    pub fn in_enlarged_target_range(position: Angle, target_position: Angle) -> bool {
-        let tolerance = Angle::new::<degree>(Self::ENLARGED_TARGET_THRESHOLD_DEGREE);
-        position > (target_position - tolerance) && position < (target_position + tolerance)
-    }
-
-    pub fn in_or_above_enlarged_target_range(position: Angle, target_position: Angle) -> bool {
-        let tolerance = Angle::new::<degree>(Self::ENLARGED_TARGET_THRESHOLD_DEGREE);
-        Self::in_enlarged_target_range(position, target_position)
-            || position >= target_position + tolerance
-    }
-}
-
 struct SlatFlapControlComputer {
     config_index_id: VariableIdentifier,
     slat_flap_system_status_word_id: VariableIdentifier,
     slat_flap_actual_position_word_id: VariableIdentifier,
 
+    powered_by: ElectricalBusType,
+    consumed_power: Power,
+    is_powered: bool,
+    is_powered_delayed: DelayedFalseLogicGate,
+
     flaps_channel: FlapsChannel,
     slats_channel: SlatsChannel,
 }
 
 impl SlatFlapControlComputer {
-    fn new(context: &mut InitContext, num: u8) -> Self {
+    // `TRANSPARENCY_TIME` is the maximum duration (200ms) the SFCC continues to
+    // output valid data after a power loss. If the power loss exceeds this time,
+    // the SFCC loses all data, zeroes its outputs, and re-initializes once power
+    // is restored.
+    const TRANSPARENCY_TIME: Duration = Duration::from_millis(200); //ms
+    const MAX_POWER_CONSUMPTION_WATT: f64 = 90.; //W Cumulative for the whole SFCC
+
+    fn new(context: &mut InitContext, num: u8, powered_by: ElectricalBusType) -> Self {
         Self {
             config_index_id: context.get_identifier("FLAPS_CONF_INDEX".to_owned()),
             slat_flap_system_status_word_id: context
                 .get_identifier(format!("SFCC_{num}_SLAT_FLAP_SYSTEM_STATUS_WORD")),
             slat_flap_actual_position_word_id: context
                 .get_identifier(format!("SFCC_{num}_SLAT_FLAP_ACTUAL_POSITION_WORD")),
 
-            flaps_channel: FlapsChannel::new(context, num),
-            slats_channel: SlatsChannel::new(context, num),
+            powered_by,
+            consumed_power: Power::new::<watt>(Self::MAX_POWER_CONSUMPTION_WATT),
+            is_powered: false,
+            is_powered_delayed: DelayedFalseLogicGate::new(Self::TRANSPARENCY_TIME)
+                .starting_as(false),
+
+            flaps_channel: FlapsChannel::new(context, num, powered_by),
+            slats_channel: SlatsChannel::new(context, num, powered_by),
```

Comment:
I don't quite understand why the channels do their own power management if they are part of the SFCC? IMO the channels themself should not worry about the power state.

Reply:
Should I pass the `is_powered` argument in the update function of each channel?

Reply:
Well,  technically, the two channels have independent powersupplies, even though the supply bus is the same. But the circuit breaker is not, so this would be relevant once we simulate CBs

Reply:
True... Should I leave as is with a note or simplify?

Reply:
I didn't know that the channels are on different circuit breakers. In this case I agree to the previous comments and we can leave it as it is now.